File: easy/2108_find_first_palindromic_string_in_the_array/solution.py
from typing import List


class Solution:
    def firstPalindrome(self, words: List[str]) -> str:
        for word in words:
            start, end = 0, len(word) - 1

            while True:
                if start > end:
                    return word
                elif word[start] != word[end]:
                    break

                start += 1
                end -= 1

        return ""

File: easy/2108_find_first_palindromic_string_in_the_array/solution.cpp
#include <string>
#include <vector>

class Solution {
public:
  std::string firstPalindrome(std::vector<std::string> &words) {
    for (const std::string &word : words) {
      if (std::equal(word.begin(), word.end(), word.rbegin()))
        return word;
    }

    return "";
  }
};
